internal/ledger/compare_test.go

```diff
@@ -0,0 +1,223 @@
+		blockNum: 1, txNum: 1, metadata: "md2",
+	},
+	{
+		namespace: "ns2", key: "k1", value: "v4",
+		blockNum: 1, txNum: 2, metadata: "md3",
+	},
+	{
+		namespace: "ns3", key: "k1", value: "v4",
+		blockNum: 2, txNum: 1, metadata: "md4",
+	},
+}
+
+var caseSamples3 []*testRecord = []*testRecord{
+	{
+		namespace: "ns1", key: "k1", value: "v1",
+		blockNum: 1, txNum: 1, metadata: "md1",
+	},
+	{
+		namespace: "ns2", key: "k1", value: "v3",
+		blockNum: 1, txNum: 2, metadata: "md3",
+	},
+	{
+		namespace: "ns3", key: "k1", value: "v4",
+		blockNum: 2, txNum: 1, metadata: "md4",
+	},
+}
+
+// Signable metadata to allow for snapshots to pass checking
+var signableMetadata1 = &kvledger.SnapshotSignableMetadata{
+	ChannelName:            "testchannel",
+	LastBlockNumber:        caseSamples1[len(caseSamples1)-1].blockNum,
+	LastBlockHashInHex:     "ba00545bfd2681a2453c26b281ea932fedf53bc40c5dd5a9091ab8df8fc4d371",
+	PreviousBlockHashInHex: "72eacca7e750be47a4bc8ead206df62336ccb4dde06c85f9d4d03a9f87414811",
+	FilesAndHashes: map[string]string{
+		"private_state_hashes.data":     "8da19e2a0fb9430dad7a5cc291dea21cb430d3449c450a910f3b738421d65ec8",
+		"private_state_hashes.metadata": "c66f7509f5b1d75d48f711b937cf0dd07db67fb335667d7369a052ae5697daf4",
+		"public_state.data":             "f2a804b048d62a662ffbac5f3dfac568bf2c8cfe35e24b93c89b96d32f5e8112",
+		"public_state.metadata":         "636e33e1a98209850a51eab6a01835386243cdf3434fd71fa3f9ed5c971ebc97",
+		"txids.data":                    "a68548bfc196d59d8ff70ddd58524c4cc03e14a03336dfe9c32c7533825dff09",
+		"txids.metadata":                "9e6282e4f25e370ce617e21d6fe265e88b9e7b8682cf00059b9d128d9381f09d",
+	},
+	StateDBType: "testdatabase",
+}
+
+var signableMetadata2 = &kvledger.SnapshotSignableMetadata{
+	ChannelName:            "testchannel",
+	LastBlockNumber:        caseSamples1[len(caseSamples1)-1].blockNum,
+	LastBlockHashInHex:     "ba00545bfd2681a2453c26b281ea932fedf53bc40c5dd5a9091ab8df8fc4d371",
+	PreviousBlockHashInHex: "72eacca7e750be47a4bc8ead206df62336ccb4dde06c85f9d4d03a9f87414811",
+	FilesAndHashes: map[string]string{
+		"private_state_hashes.data":     "8da19e2a0fb9430dad7a5cc291dea21cb430d3449c450a910f3b738421d65ec8",
+		"private_state_hashes.metadata": "c66f7509f5b1d75d48f711b937cf0dd07db67fb335667d7369a052ae5697daf4",
+		"public_state.data":             "8d77cdbbd2f6d9f7962ce0ea4346de2224e783cb35f5cd15e3d8add69ecc584a",
+		"public_state.metadata":         "3f4737b4bfd91ffc166aca4eadbd6cd07ce6cacfadfdb6405ab38419dde3b217",
+		"txids.data":                    "ca3d6a796929d57e0f276df5d31cff20eb48bdc1192afa0c79cc0b29fa870397",
+		"txids.metadata":                "ca6c6588fa01171b200740344d354e8548b7470061fb32a34f4feee470ec281f",
+	},
+	StateDBType: "testdatabase",
+}
+
+const (
+	expectedDifferenceResult = "[\n{\"namespace\":\"ns2\",\"key\":\"k1\",\"snapshotrecord1\":{\"value\":\"v3\",\"blockNum\":1,\"txNum\":2},\"snapshotrecord2\":{\"value\":\"v4\",\"blockNum\":1,\"txNum\":2}}\n]\n"
```

Comment:
```suggestion
	expectedDifferenceResult = `[
		{ "namespace":"ns2",
		  "key":"k1",
		  "snapshotrecord1":{
			  "value":"v3",
			  "blockNum":1,
			  "txNum":2
		  },
		  "snapshotrecord2":{
			  "value":"v4",
			  "blockNum":1,
			  "txNum":2
			}
		}
	]`
```
This way, it's more readable. 
When you compare down below, instead of `require.Equal` use `require.JSONEq`